src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -102,8 +111,14 @@ def self.getModules(env)
       # NOTE: If in the future we want to collect more
       #       Module settings, this should be more programatic
       #       rather than getting these settings one by one
-      module_data[:name] = mod.forge_name
-      module_data[:version] = mod.version
+      if mod.forge_name
+        module_data[:name] = mod.forge_name
+        module_data[:version] = mod.version
```

Comment:
Yeah I'm guessing a lot breaks because of this. I just tried using the module tool to uninstall those modules I was using to test this PR with, and it gave me the same exception/stacktrace:

```
brian@localhorse:ntp % sudo puppet module uninstall puppetlabs-ntp;sudo puppet module uninstall puppetlabs-stdlib
Password:
Error: No version module metadata provided for ntp
Notice: Preparing to uninstall 'puppetlabs-ntp' ...
Error: No version module metadata provided for ntp
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module.rb:160:in `block in load_metadata'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module.rb:158:in `each'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module.rb:158:in `load_metadata'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module.rb:69:in `initialize'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:358:in `new'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:358:in `block (3 levels) in modules_by_path'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:357:in `map'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:357:in `block (2 levels) in modules_by_path'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:353:in `chdir'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:353:in `block in modules_by_path'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:351:in `each'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:351:in `modules_by_path'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module_tool/applications/uninstaller.rb:51:in `find_installed_module'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module_tool/applications/uninstaller.rb:24:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module_tool/applications/application.rb:12:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/face/module/uninstall.rb:63:in `block (3 levels) in <top (required)>'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/interface/action.rb+eval[wrapper]:242:in `uninstall'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/face_base.rb:248:in `main'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:352:in `run_command'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:344:in `block in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util.rb:540:in `exit_on_fail'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:344:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:132:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:72:in `execute'
/usr/local/bin/puppet:5:in `<main>'
Error: No version module metadata provided for ntp
Notice: Preparing to uninstall 'puppetlabs-stdlib' ...
Error: No version module metadata provided for ntp
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module.rb:160:in `block in load_metadata'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module.rb:158:in `each'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module.rb:158:in `load_metadata'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module.rb:69:in `initialize'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:358:in `new'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:358:in `block (3 levels) in modules_by_path'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:357:in `map'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:357:in `block (2 levels) in modules_by_path'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:353:in `chdir'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:353:in `block in modules_by_path'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:351:in `each'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/node/environment.rb:351:in `modules_by_path'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module_tool/applications/uninstaller.rb:51:in `find_installed_module'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module_tool/applications/uninstaller.rb:24:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/module_tool/applications/application.rb:12:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/face/module/uninstall.rb:63:in `block (3 levels) in <top (required)>'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/interface/action.rb+eval[wrapper]:242:in `uninstall'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application/face_base.rb:248:in `main'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:352:in `run_command'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:344:in `block in run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util.rb:540:in `exit_on_fail'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/application.rb:344:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:132:in `run'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/command_line.rb:72:in `execute'
/usr/local/bin/puppet:5:in `<main>'
```